src/main/java/uk/gov/hmcts/ccd/endpoint/exceptions/RestExceptionHandler.java

```diff
@@ -138,30 +136,59 @@ protected ResponseEntity<Object> handleMethodArgumentNotValid(MethodArgumentNotV
             .body(error);
     }
 
-    private HttpStatus getHttpStatus(Throwable causeOfException) {
-        HttpStatus httpStatus = null;
-        if (causeOfException instanceof HttpServerErrorException) {
-            httpStatus = ((HttpServerErrorException) causeOfException).getStatusCode();
-            if (httpStatus == HttpStatus.INTERNAL_SERVER_ERROR) {
-                httpStatus = HttpStatus.BAD_GATEWAY;
-            }
-        } else if (causeOfException instanceof FeignException.FeignServerException) {
-            httpStatus = HttpStatus.valueOf(((FeignException) causeOfException).status());
-            if (httpStatus == HttpStatus.INTERNAL_SERVER_ERROR) {
-                httpStatus = HttpStatus.BAD_GATEWAY;
-            }
-        } else if (causeOfException instanceof HttpClientErrorException) {
-            httpStatus = ((HttpClientErrorException) causeOfException).getStatusCode();
-            if (httpStatus != HttpStatus.UNAUTHORIZED) {
-                httpStatus = HttpStatus.INTERNAL_SERVER_ERROR;
-            }
-        } else if (causeOfException instanceof FeignException.FeignClientException) {
-            httpStatus = HttpStatus.valueOf(((FeignException) causeOfException).status());
-            if (httpStatus != HttpStatus.UNAUTHORIZED) {
-                httpStatus = HttpStatus.INTERNAL_SERVER_ERROR;
+    private HttpStatus getHttpStatus(final Throwable causeOfException) {
+        int httpStatusCode = checkAndRetrieveStatusCode(causeOfException);
+        if (httpStatusCode > 0) {
+            return HttpStatus.valueOf(assignHttpCode(httpStatusCode));
+        }
+
+        return checkReadTimeoutException(causeOfException);
+    }
+
+    private int checkAndRetrieveStatusCode(final Throwable causeOfException) {
+        int httpStatusCode = 0;
+        if (causeOfException instanceof HttpStatusCodeException) {
+            httpStatusCode = ((HttpStatusCodeException) causeOfException).getRawStatusCode();
+        } else if (causeOfException instanceof FeignException) {
+            httpStatusCode = ((FeignException) causeOfException).status();
+        }
+        return httpStatusCode;
+    }
+
+    // return 502 for 500 status code,other 5xx will be kept as is
+    // return 401 for 401 status code, other 4xx will return 500
+    private int assignHttpCode(final int httpCode) {
```

Comment:
Might be better if this method uses the HttpStatus enum values rather than the integer equivalents, e.g. HttpStatus.UNAUTHORIZED rather than 401.

Reply:
Thanks Paul, I have made the change accordingly.

Reply:
Changes look good.